src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorFactory.java

```diff
@@ -0,0 +1,215 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.jazzer.mutation.mutator.collection;
+
+import static com.code_intelligence.jazzer.mutation.support.TypeSupport.parameterTypesIfParameterized;
+import static java.lang.Math.min;
+import static java.util.stream.Collectors.toMap;
+
+import com.code_intelligence.jazzer.mutation.annotation.WithSize;
+import com.code_intelligence.jazzer.mutation.api.*;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.AnnotatedType;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+final class MapMutatorFactory extends MutatorFactory {
+  @Override
+  public Optional<SerializingMutator<?>> tryCreate(AnnotatedType type, MutatorFactory factory) {
+    // noinspection OptionalIsPresent
+    return parameterTypesIfParameterized(type, Map.class)
+        .map(parameterTypes
+            -> parameterTypes.stream()
+                   .map(factory::tryCreate)
+                   .flatMap(o -> o.isPresent() ? Stream.of(o.get()) : Stream.empty())
+                   .collect(Collectors.toList()))
+        .filter(elementMutators -> elementMutators.size() == 2)
+        .map(elementMutators -> {
+          int min = MapMutator.DEFAULT_MIN_SIZE;
+          int max = MapMutator.DEFAULT_MAX_SIZE;
+          Optional<WithSize> withSize = Arrays.stream(type.getAnnotations())
+                                            .filter(annotation -> annotation instanceof WithSize)
+                                            .findFirst()
+                                            .map(annotation -> (WithSize) annotation);
+          if (withSize.isPresent()) {
+            min = withSize.get().min();
+            max = withSize.get().max();
+          }
+          return new MapMutator<>(elementMutators.get(0), elementMutators.get(1), min, max);
+        });
+  }
+
+  private static final class MapMutator<K, V> extends SerializingInPlaceMutator<Map<K, V>> {
+    private static final int DEFAULT_MIN_SIZE = 0;
+    private static final int DEFAULT_MAX_SIZE = 1000;
+
+    private final SerializingMutator<K> keyMutator;
+    private final SerializingMutator<V> valueMutator;
+    private final int minSize;
+    private final int maxSize;
+
+    MapMutator(SerializingMutator<K> keyMutator, SerializingMutator<V> valueMutator, int minSize,
+        int maxSize) {
+      this.keyMutator = keyMutator;
+      this.valueMutator = valueMutator;
+      this.minSize = minSize;
+      this.maxSize = maxSize;
+    }
+
+    @Override
+    public Map<K, V> read(DataInputStream in) throws IOException {
+      int size = Math.max(in.readInt(), 0);
```

Comment:
What if the user changes the code, and a large number that was interpreted as an integer in previous runs, suddenly becomes the size of a map/list? We aren't saving mutator IDs in the corpus, right?

Reply:
Added size checks.